### PR TITLE
Drop rustc-serialize dependency

### LINKED ISSUE #3944: Drop Decodable dependency for call_main_without_stdin
```rust
pub fn call_main_without_stdin<Flags: Decodable>(
            exec: fn(Flags, &Config) -> CliResult,
            config: &Config,
            usage: &str,
            args: &[String],
            options_first: bool) -> CliResult
```

This may be blocked on https://github.com/docopt/docopt.rs/issues/128 unless we replace docopt.

### PR DESCRIPTION
**Pull request by ordian**
Fixes #3944.

### CODE REVIEW

**Comment by alexcrichton** [comment]
💯 

@bors: r+

Thanks so much!

**Comment by ordian** [comment]
Why should "cargo version" work [test](https://github.com/rust-lang/cargo/blob/master/tests/version.rs#L14), there is no such command - only option? On the other hand, I don't know why deserialize_unit is called instead of printing usage.

File: src/bin/version.rs [hunk 1]
```diff
@@ -3,8 +3,8 @@ use std::env;
 use cargo;
 use cargo::util::{CliResult, Config};
 
-#[derive(RustcDecodable)]
-pub struct Options;
+#[derive(Deserialize)]
+pub struct Options {}
```
Comment (ordian) [inline src/bin/version.rs]: Docopt currently doesn't support unit structs, as a workaround I changed it to empty struct.

Comment (ordian) [inline src/bin/version.rs]: [PR](https://github.com/docopt/docopt.rs/pull/217) just merged.

**Comment by ordian** [comment]
@alexcrichton Should we wait for new docopt release with [fix](https://github.com/docopt/docopt.rs/pull/217) or is it ok to use suggested workaround?

**Comment by alexcrichton** [comment]
@bors: r+

Nah the workaround sounds fine by me!

**Pull request merged**